ui/lib/core/addon/components/kv-object-editor.js

```diff
@@ -55,7 +56,10 @@ export default class KvObjectEditor extends Component {
   @action
   createKvData(elem, [value]) {
     this.kvData = KVObject.create({ content: [] }).fromJSON(value);
-    this.addRow();
+
```

Comment:
What about just returning if this.args.isSingleRow?
`if (this.args.isSingleRow) return`

Reply:
We still want to add a single row on initial render (if it doesn't exist already). For instance, when we're editing a message with existing kv values, we don't want to add a row, however, if we're creating a new message, we want to add a row. Let me know if that makes sense!

Reply:
Ah I see. I think the reverse logic is a little easier to follow `if (this.args.isSingleRow && value) return`, but either works! I'll defer to you. Feel free to leave as is :)